tests/test_docker_wrapper.py

```diff
@@ -599,3 +623,81 @@ def _patch_docker_wrapper_dependencies(self, is_numa_enabled, cpu_info, node_mem
                              return_value=node_mem
                              ), mock.patch.object(docker_wrapper, 'open', new=m):
             yield
+
+    @pytest.yield_fixture
+    def mock_mac_address(self):
+        with mock.patch.object(
+                docker_wrapper.paasta_tools.mac_address,
+                'reserve_unique_mac_address',
+                return_value=('00:00:00:00:00:00', None),
+        ) as mock_mac_address:
+            yield mock_mac_address
+
+    def test_mac_address(self, mock_mac_address, mock_execlp):
+        argv = [
+            'docker',
+            'run',
+            '--env=PAASTA_FIREWALL=1',
+        ]
+        docker_wrapper.main(argv)
+
+        assert mock_execlp.mock_calls == [mock.call(
+            'docker',
+            'docker',
+            'run',
+            '--mac-address=00:00:00:00:00:00',
+            '--env=PAASTA_FIREWALL=1',
+        )]
+
+    def test_mac_address_not_run(self, mock_mac_address, mock_execlp):
+        argv = [
+            'docker',
+            'inspect',
+            '--env=PAASTA_FIREWALL=1',
+        ]
+        docker_wrapper.main(argv)
+
+        assert mock_mac_address.call_count == 0
+        assert mock_execlp.mock_calls == [mock.call(
+            'docker',
+            'docker',
+            'inspect',
+            '--env=PAASTA_FIREWALL=1',
+        )]
+
+    def test_mac_address_already_set(self, mock_mac_address, mock_execlp):
+        argv = [
+            'docker',
+            'run',
+            '--mac-address=12:34:56:78:90:ab',
+            '--env=PAASTA_FIREWALL=1',
+        ]
+        docker_wrapper.main(argv)
+
+        assert mock_mac_address.call_count == 0
+        assert mock_execlp.mock_calls == [mock.call(
+            'docker',
+            'docker',
+            'run',
+            '--mac-address=12:34:56:78:90:ab',
+            '--env=PAASTA_FIREWALL=1',
+        )]
+
+    @mock.patch('sys.stderr', new_callable=six.StringIO, autospec=None)
+    def test_mac_address_no_lockdir(self, mock_stderr, mock_execlp, tmpdir):
```

Comment:
there's a `capsys` pytest fixture which might be slightly cleaner than mocking sys.stderr: https://docs.pytest.org/en/latest/capture.html